File: bin/contig_lengths_hist.pl
#TITLE: contig_lengths_hist.pl
#AUTHOR: BRETT ROBERTS
#DATE LAST MODIFIED: 03/02/13
#PURPOSE: create csv file with contig names and lengths from fasta file

my $fasta = $ARGV[0];
my $output = $ARGV[1];
open FASTA, "<$fasta";

while (my $line = <FASTA>) {                 
    chomp $line;
    if ($line =~ /^>/) {
        my @line = split(/\s+/, $line);
        $contig = $line[0];
        $contig =~ s/>//;
    }
    else{          
        $sequence = $line;
        $sequence =~ s/\s+//;
        $length = length($sequence);
        $sequences{$contig} += $length;
    }
}

open OUT, ">$output";
print OUT "contig,length\n";
foreach my $contig (keys %sequences) {
    print OUT "$contig,$sequences{$contig}\n";
}

File: bin/contig_lengths_hist.R
#TITLE: contig_length_hist.R
#AUTHOR: Brett Roberts
#DATE LAST MODIFIED: 11/07/12
#PURPOSE: creates a histogram showing the distribution of contig and scaffold lengths in the two assemblies

input_dir="/Users/robotien/Desktop/Brett/SG_Biofuels/dilara/sspace/"
output_dir="/Users/robotien/Desktop/Brett/SG_Biofuels/dilara/sspace/"

setwd(input_dir)
gap_lengths <-read.csv("run5_gaplengths.txt", header=TRUE)

setwd(output_dir)

hist(gap_lengths$length,breaks=seq(from=0,to=10000, by=500),col="grey", xlab="Gap Length", main="SSPACE Run5 Gap Lengths",ylim=c(1,900))


